Automattic/wp-calypso
# Packages: Introduce `@automattic/is-plain-object`

#### Proposed Changes

This PR introduces a new `@automattic/is-plain-object` package that is intended to be public, and used both in Calypso and in Gutenberg as a replacement for Lodash's `isPlainObject()` function. 

See https://github.com/WordPress/gutenberg/pull/42471#issuecomment-1186879992 for the motivation for this package. Essentially, we're aiming to minimize the creation of new packages as Lodash replacements, but we will still end up having a few of them for the greater good.

#### Testing Instructions

* Verify all package tests pass: `yarn test-packages packages/is-plain-object`
* Verify all state tests pass: `yarn test-client state`
* Verify all checks are green and there are no TS errors/warnings introduced.
* Verify the preferences helper still works (displayed when you roll over the environment name in Calypso on non-prod environments).

#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
